src/main/java/com/artipie/RepoPermissions.java

```diff
@@ -110,8 +118,37 @@ public FromSettings(final Settings settings) {
         }
 
         @Override
-        public CompletionStage<Map<String, List<String>>> get(final String repo) {
-            throw new UnsupportedOperationException("To implemented later");
+        public CompletionStage<Map<String, List<String>>> permissions(final String repo) {
+            return this.yaml(new Key.From(String.format("%s.yaml", repo))).thenApply(
+                yaml -> Optional.ofNullable(yaml.yamlMapping("repo").yamlMapping("permissions"))
+                .map(
+                    perms -> {
+                        final Map<String, List<String>> res = new HashMap<>();
+                        perms.keys().forEach(
+                            node -> res.put(
+                                node.asScalar().value(),
+                                perms.yamlSequence(node.asScalar().value()).values().stream()
+                                .map(item -> Scalar.class.cast(item).value())
+                                    .collect(Collectors.toList())
+                            )
+                        );
+                        return res;
+                    }
+                ).orElse(Collections.emptyMap())
+            );
+        }
+
+        /**
+         * Credentials as yaml.
+         * @param key Repo settings key
+         * @return Completion action with yaml
+         */
+        private CompletionStage<YamlMapping> yaml(final Key key) {
+            return new RxStorageWrapper(this.storage())
+                .value(key)
+                .to(ContentAs.YAML)
+                .to(SingleInterop.get())
+                .thenApply(yaml -> (YamlMapping) yaml);
```

Comment:
@olenagerasimova I'd suggest to remove dynamic casting and change return type to `<? extends YamlMapping>` or specify type param for `to` method as `<YamlMapping>`